python/cudf/cudf/core/column/categorical.py

```diff
@@ -582,7 +595,30 @@ def is_monotonic_decreasing(self):
         return self._is_monotonic_decreasing
 
     def as_categorical_column(self, dtype, **kwargs):
-        return self
+        if isinstance(dtype, str) and dtype == "category":
+            return self
+        if (
+            isinstance(
+                dtype, (cudf.core.dtypes.CategoricalDtype, pd.CategoricalDtype)
+            )
+            and (dtype.categories is None)
+            and (dtype.ordered is None)
+        ):
+            return self
+
+        if isinstance(dtype, pd.CategoricalDtype):
+            dtype = CategoricalDtype(
+                categories=dtype.categories, ordered=dtype.ordered
+            )
+
+        if not isinstance(self.categories, type(dtype.categories._values)):
```

Comment:
Would any strangeness occurs here if the type returned was a CuPy array ?  If not, then disregard

Reply:
Nope, won't be a possibility. Because by the time code-flow hit's this line `dtype` has to be `cudf.CatgeoricalDtype`. and `.categories._values` actually represents a column.

example:
```python
>>> s = cudf.Series([1,2,], dtype='category')
>>> s.dtype
<cudf.core.dtypes.CategoricalDtype object at 0x7f855c861f10>
>>> s.dtype.categories
Int64Index([1, 2], dtype='int64')
>>> s.dtype.categories._values
<cudf.core.column.numerical.NumericalColumn object at 0x7f855ae2dc20>
```